### DISCUSSION TITLE
Side Drawer

### DISCUSSION THREAD

**Discussion by annefranca** [Telefonica/mistica-design#1835] [Proposals]
### Goal

Discutir e avaliar a proposta de adicionar o componente Side Drawer em Mística.

### Context

Como conversado em nossa reunião semanal, começamos a desenvolver o Side Drawer na biblioteca Extended para o projeto Web Essential. O componente nasceu como Extended pelo prazo que tínhamos v/s o lançamento do MVP, com isso, gostaríamos de avaliar a possibilidade do Side Drawer pertencer a biblioteca Mística.

### Impact

Ao adicionar o Side Drawer padronizado como componente melhoraria a experiência da pessoa usuária, o tempo dos times de desenvolvimento e a consistência nas aplicações das jornadas e nos fluxos.

### Discussion Questions

ND

### Do you have a visual proposal?

https://www.figma.com/design/VL1iNAjAjyTk3EIaaDmCW4/Side-Drawer?node-id=8902-1250&t=8y9xW4hnYHBSzkZh-1

**Comment by HectorSancho** [comment]
Hola Anne!

Un par de cosas que nos gustaría saber del componente que no vemos en la propuesta visual:

· ¿Tenéis alguna propuesta del responsive con su adaptación a mobile?
· ¿Pensáis usarlo para procesos de varios pasos? Si tenéis alguna referencia de UX, estaría bien verla.

Hablando entre nosotros, hemos estado discutiendo sobre como podría ser esa versión móvil. Por defecto quizás haya que pensar proporcionar algo parecido a una sheet, el hueco lateral si mantenemos la misma UI de desktop no va a funcionar muy bien.

Si usáis este componente para flujos largos de más de 2 o 3 pasos, la solución en esos casos podría ir por hacer una adaptación custom y utilizar una modal view en mobile para ese tipo de flujos full screen + navigation bar con el cerrar a la derecha).

**Reply by annefranca** [reply]
Olá!

**Sobre a adaptação para celular:** existe um estudo em andamento sobre responsive layout para breakpoint small (mobile). Estamos fazendo um bench nessa página https://www.figma.com/design/VL1iNAjAjyTk3EIaaDmCW4/Side-Drawer?node-id=9437-7263&t=bCYo0tEHemny7s42-1

Caso você tenha alguma contribuição, pode adicionar nessa página "estudos".

**Sobre a navegação:** vai ser considerada apenas para os fluxos do projeto Web Essential. Após o lançamento do MVP, vamos considerar uma nova etapa de estudo para essa funcionalidade ser refatorada, com isso, para esse entregável, a propriedade navigation é considerada como opcional.

**Comment by yceballost** [comment]
Gracias por la info Anne! Lo añadimos al backlog para que se incluya en Mística! Recuerda hablar con Julio para incluir esto para Q4 si es algo prioritario
